src/algorithms/MathUtil.java

```diff
@@ -37,10 +41,20 @@ public static int pow(int number, int pow){
      * @param n the number factorial of which must be calculated
      * @return calculated number
      */
+
     public static int factorial(int n) {
-       return 0;
+        int result = 1;
+
+        for (int i = 2; i <= n; i++) {
+            result *= i;
+        }
+
+        return result;
```

Comment:
done.

You can also write a recursive method

Reply:
public static int factorial(int n){
        int fact;
        if (n == 0)
            return 1;
        else {
            fact = n * factorial(n-1) ;
            return fact;
        }
    }    
I try, but something do not work

Reply:
it works correctly, you can optimize the code like this

if (n == 0) {
            return 1;
        }
        else
            return(n * factorial(n-1));